lib/src/working_copy.rs

```diff
@@ -1009,12 +1009,14 @@ impl TreeState {
                     let id = self.write_file_to_store(repo_path, disk_path)?;
                     // On Windows, we preserve the executable bit from the current tree.
                     #[cfg(windows)]
-                    let executable =
+                    let executable = {
+                        let _ = executable; // use the variable
```

Comment:
nit: `let () = executable;` to clarify we aren't discarding the value?

Reply:
Done here and a few lines up.